dataset.md

```diff
@@ -0,0 +1,56 @@
+# video-embedding-dataset
+
+In this document we define a dataset format for training contrastive language-video temporal aggregator models. This is a very early-stage project so please feel free to propose ideas/changes.
+
+## Overview
+
+The data processing pipeline for video-embedding-datasets is: raw dataset -> organized dataset (some common format so clip-video-encode handle it nicely) -> webdataset
+
+
+## Raw dataset
+
+You can find a list of raw datasets in:
+https://docs.google.com/document/d/12zYnjZabR2e17vPO2XpctIf1qUQeEX7kYC8GdDqWM-k/edit
+
+## Processed datasets
+
+We us the [webdataset](https://github.com/webdataset/webdataset) format therefore the final processed dataset should be in the form of a list of tar files split up into train/val/test splits. Each tar file should have:
+* numpy array (.npy) of shape (frame_count, embed_dim) where embed_dim is 512 for now
+* text caption (.cap) that describes the video
+* json file with any additional metadata (YouTube video ID, time window of video clip from larger video, etc.)
+
+
+```
+video-embedding-dataset
+ ├── train
+ │   ├── ds_00000.tar
+ |   |     ├── name0.npy
+ |   |     ├── name0.cap
+ |   |     ├── name0.json
+ |   |     ├── name1.npy
+ |   |     ├── name1.cap
+ |   |     ├── name1.json
+ |   |     └── ...
+ │   ├── ds_00001.tar
+ |   |     ├── name0.npy
```

Comment:
I advise you use unique keys over the whole dataset. That makes it possible to reshuffle the shards if needed.

An easy way to do this is to take as key (shardid*10000 + sample id) with 0 padding of 9

Reply:
Good call, in that case maybe I'll restructure the dataset so that it's just n tar files and we can have a CSV file describing which tars belong to which splits. What do you think?

Reply:
Yeah that might be a good idea indeed. It will make things simpler